src/cameras/OrthographicCamera.js

```diff
@@ -38,19 +40,51 @@ THREE.OrthographicCamera.prototype = Object.assign( Object.create( THREE.Camera.
 		this.far = source.far;
 
 		this.zoom = source.zoom;
+		this.view = source.view === null ? null : Object.assign( {}, source.view );
 
 		return this;
 
 	},
 
+	setViewOffset: function( fullWidth, fullHeight, x, y, width, height ) {
+
+		this.view = {
+			fullWidth: fullWidth,
+			fullHeight: fullHeight,
+			offsetX: x,
+			offsetY: y,
+			width: width,
+			height: height
+		};
+
+		this.updateProjectionMatrix();
+
+	},
+
 	updateProjectionMatrix: function () {
 
 		var dx = ( this.right - this.left ) / ( 2 * this.zoom );
 		var dy = ( this.top - this.bottom ) / ( 2 * this.zoom );
 		var cx = ( this.right + this.left ) / 2;
 		var cy = ( this.top + this.bottom ) / 2;
 
-		this.projectionMatrix.makeOrthographic( cx - dx, cx + dx, cy + dy, cy - dy, this.near, this.far );
+		var left = cx - dx;
+		var right = cx + dx;
+		var top = cy + dy;
+		var bottom = cy - dy;
+
+		if ( this.view !== null ) {
+
+			var zoom = this.zoom / ( this.view.width / this.view.fullWidth );
```

Comment:
I think this should simply be:

    var zoom = this.zoom;

In fact, it is not clear to me the purpose of `fullWidth`. And `fullHeight` is not used at all.